### PR TITLE
Cst compat

### PR DESCRIPTION
**Pull request by ZacLN**
Amendments for CSTParser changes (https://github.com/julia-vscode/CSTParser.jl/pull/190)

### CODE REVIEW

File: src/requests/textdocument.jl [hunk 1]
```diff
@@ -68,130 +68,17 @@ end
 
 
 function textDocument_didChange_notification(params::DidChangeTextDocumentParams, server::LanguageServerInstance, conn)
+    t=  time()
```
Comment (pfitzseb) [inline src/requests/textdocument.jl:71]:
```suggestion

```

File: src/runserver.jl [hunk 2]
```diff
@@ -49,3 +49,10 @@ end
 choose_env() = something(get(ARGS, 1, nothing),          # 1. path passed explicitly
                          Base.current_project(pwd()),    # 2. parent project of pwd()
                          Base.load_path_expand("@v#.#")) # 3. default "global" env
+
+function choose()
+    maybe_dirname = x -> x !== nothing ? dirname(x) : nothing
+    something(get(ARGS, 1, nothing),                            # 1. path passed explicitlysomething(get(ARGS, 1, nothing))
```
Comment (pfitzseb) [inline src/runserver.jl:55]:
```suggestion
    something(get(ARGS, 1, nothing),                            # 1. path passed explicitly
```

File: src/requests/init.jl [hunk 3]
```diff
@@ -24,7 +24,7 @@ const serverCapabilities = ServerCapabilities(
     true,
     false,
     ExecuteCommandOptions(missing, collect(keys(LSActions))),
-    false,
+    true,
```
Comment (pfitzseb) [inline src/requests/init.jl:27]: This is `selectionRangeProvider`, right? Did we always support that? If so, why was it disabled?

**Review by pfitzseb — COMMENTED** [review]
Let's turn incremental sync back on and see what happens, at least in the next insiders release? I kinda hope that all the code churn here and in CSTParser might've fixed the issues we had.

File: src/requests/init.jl (hunk 3 above)
Comment (ZacLN) [reply src/requests/init.jl:27]: I must have been fiddling with something, will turn off

**Comment by ZacLN** [comment]
@julia-vscode/core could someone with a windows computer try to get this passing? Debugging by pushing to CI runs is a nightmare.

The issue is with path handling on older Julia verions (<1.1 I think, possibly <1.3). Relevant code is https://github.com/julia-vscode/LanguageServer.jl/blob/master/src/utilities.jl#L286:L309 or there abouts

File: src/requests/init.jl [hunk 4]
```diff
@@ -1,6 +1,6 @@
 const serverCapabilities = ServerCapabilities(
     TextDocumentSyncOptions(true,
-    TextDocumentSyncKinds.Full,
+    TextDocumentSyncKinds.Incremental,
```
Comment (pfitzseb) [inline src/requests/init.jl:3]: Let's try re-enabling this?

**Review by pfitzseb — APPROVED** [review]

**Pull request merged**